cinema/views.py

```diff
@@ -45,9 +76,31 @@
 
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
-    queryset = MovieSession.objects.all()
+    queryset = (
+        MovieSession.objects
+        .select_related("movie", "cinema_hall")
+        .annotate(
+            tickets_available=(
+                F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets") - 1
```

Comment:
The calculation of `tickets_available` subtracts an extra 1, which might lead to incorrect counts. Verify if this subtraction is intentional and correct.